### PR TITLE
Extract `ActionController::Helpers`

### PR DESCRIPTION
**Pull request by KaanOzkan**
resolves https://github.com/Shopify/sorbet-issues/issues/186

### CODE REVIEW

File: lib/tapioca/compilers/dsl/action_mailer.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ module Dsl
       #
       # ~~~rbi
       # # notifier_mailer.rbi
-      # # typed: true
+      # # typed: strong
```
[the remark this answers is not in the document]
Comment (KaanOzkan) [reply lib/tapioca/compilers/dsl/action_mailer.rb:32]: I was just keeping it consistent with what we expect in the tests. Generators are producing `#typed: strong`

If the pipeline is taking care of generating `typed: true` documentation can just say `typed: true` then

File: spec/tapioca/compilers/dsl/action_controller_helpers_spec.rb [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        class UserController
+          sig { returns(UserController::HelperProxy) }
+          def helpers; end
+        end
+
+        module UserController::HelperMethods
+          sig { params(user: T.untyped).returns(T.untyped) }
+          def greet(user); end
+
+          sig { params(user_id: Integer).returns(<VOID>) }
```
Comment (paracycle) [inline spec/tapioca/compilers/dsl/action_controller_helpers_spec.rb:186]: Bah, we need to fix that. 😞 Looking into it

Comment (paracycle) [reply spec/tapioca/compilers/dsl/action_controller_helpers_spec.rb]: Fixed

File: lib/tapioca/compilers/dsl/action_controller_helpers.rb [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+      # end
+      #
+      # module UserController::HelperMethods
+      #    include MyHelper
+      #
+      #    sig { params(user: T.untyped).returns(T.untyped) }
+      #    def age(user); end
+      #
+      #    sig { params(args: T.untyped, blk: T.untyped).returns(T.untyped) }
+      #    def current_user_name(*args, &blk); end
```
Comment (paracycle) [inline lib/tapioca/compilers/dsl/action_controller_helpers.rb:63]:
```suggestion
      #    sig { returns(T.untyped) }
      #    def current_user_name; end
```

Comment (KaanOzkan) [reply lib/tapioca/compilers/dsl/action_controller_helpers.rb:63]: My bad forgot about it. Good catch.

**Pull request merged**